jsonparse.js

```diff
@@ -408,6 +415,34 @@ proto.onToken = function (token, value) {
   }
 };
 
+/**
+ * determines if precision is lost when parsing the JSON representation of a number to a javascript number
+ * @param {string} jsonNumber
+ * @param {number} javascriptNumber
+ * @return {boolean}
+ */
+function precisionWasLost (jsonNumber, javascriptNumber) {
+  return getDigitsSpecifyingPrecision(jsonNumber) !== getDigitsSpecifyingPrecision(JSON.stringify(javascriptNumber))
+}
+
+/**
+ * Takes in a string represention of a number and returns a simplified version of the significand
+ * as a string.
+ *
+ * The simplified significand only contains the digits and we remove any trailing zeros or decimal
+ * information.
+ * @param {string} numberString
+ * @return {string}
+ */
+function getDigitsSpecifyingPrecision(numberString) {
```

Comment:
```suggestion
function getSignificand(numberString) {
```

Reply:
Can we leave a link to the wikipedia page

Reply:
Can we clarify the assumed shape of this input? For example, we are assuming that if the number has any `., it has at most 1. 

It might just be "a number following the JSON specification"